src/contests/ContestShow.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+
+export interface ContestShowProps {
+  name: string;
+  description: string;
+  joined: boolean;
+  startAt: Date;
+  endAt: Date;
+  onJoinButtonClick: () => void;
+}
+
+export default class ContestShow extends React.Component<ContestShowProps, {}> {
+  constructor(props: ContestShowProps) {
+    super(props);
+    this.state = {
+      initialized: false
+    };
+  }
+
+  private renderButton() {
+    const now = new Date();
+    if (!this.props.joined && now < this.props.endAt && now > this.props.startAt) {
+      return (<div className='center'><button className='btn'>参加する</button></div>);
+    }
+  }
+
+  public render() {
+    return (
+      <div className='contest-main'>
+        <div> 
+          <div className='contest-header'> <span className='contest-title'>{ this.props.name }</span> </div>
```

Comment:
I think this line should be split into new lines.

```
<div className='contest-header'>
  <span className='contest-title'>{ this.props.name }</span>
</div>
```

(This is trivial thing)